airflow/bin/cli.py

```diff
@@ -312,6 +327,13 @@ def run(args, dag=None):
 
     ti = TaskInstance(task, args.execution_date)
 
+    # If task uses impersonation, give write perm to log file to the user
+    if ti.run_as_user and not args.cfg_path:
+        uid = getpwnam(ti.run_as_user).pw_uid
+        fd = os.open(filename, os.O_RDWR)
+        os.fchown(fd, uid, -1)
```

Comment:
I really dont like this approach. Basically you are circumventing the admins configuration by disallowing some users to write to certain directories. This seems to me like a security issue waiting to happen. Logging in case of run_as really needs to happen to a service.